### PR TITLE
feat: Chat UI - Emoji Search Modal & Reaction Overlay

### PR DESCRIPTION
**Pull request by ice-kreios**
## Description
Implements a searchable emoji picker modal with recent emoji reactions support and the MessageReactionDialog widget, a new modal component that displays messages with reaction controls. This feature enhances user interaction by allowing users to view and react to messages in a dedicated dialog view with a blurred backdrop.

## Additional Notes
- Implements adaptive positioning based on screen constraints
- Uses MediaService for message content capture
- Handles both sender and receiver message layouts
- Search functionality for quick emoji lookup
- Categorized emoji groups displayed in a grid layout
- Automatic saving of selected emojis to recent reactions

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactoring
- [ ] Documentation
- [ ] Chore

## Screenshots (if applicable)
image 
image 
image 
image 
image 

## Figma Link (if applicable)
https://www.figma.com/design/3TnZngoFeklOVC8TexlR1D/%F0%9F%A7%8A-ICE-Cloud?node-id=45664-166114&m=dev

### CODE REVIEW

File: lib/app/components/overlay_menu/components/overlay_menu_item.dart [hunk 1]
```diff
@@ -10,13 +10,16 @@ class OverlayMenuItem extends StatelessWidget {
...
+    this.minWidth = 99.0,
   });
 
   final String label;
   final Widget icon;
   final VoidCallback onPressed;
   final Color? labelColor;
-
+  final double verticalPadding;
+  final double minWidth;
```
[the remark this answers is not in the document]
Comment (ice-kreios) [reply lib/app/components/overlay_menu/components/overlay_menu_item.dart]: fixed

File: lib/app/features/chat/messages/views/components/message_reaction_dialog/components/message_reaction_emoji_bar.dart [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import 'package:ion/app/features/chat/providers/recent_emoji_reactions_provider.dart';
+import 'package:ion/app/router/app_routes.dart';
+import 'package:ion/generated/assets.gen.dart';
+
+class MessageReactionEmojiBar extends ConsumerWidget {
+  const MessageReactionEmojiBar({
+    required this.isMe,
+    super.key,
+  });
+  static final height = 72.0.s;
```
[the remark this answers is not in the document]
Comment (ice-kreios) [reply lib/app/features/chat/messages/views/components/message_reaction_dialog/components/message_reaction_emoji_bar.dart]: fixed

**Review by ice-alcides — APPROVED** [review]

File: lib/app/features/chat/messages/views/components/message_reaction_dialog/message_reaction_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        Positioned(
+          left: isMe ? null : ScreenSideOffset.defaultSmallMargin,
+          right: isMe ? ScreenSideOffset.defaultSmallMargin : null,
+          // top: overflowBottomSize < 0 ? null : position.dy - emojiSectionHeight
+          top: topY,
+          bottom: overflowBottomSize < 0
+              ? MediaQuery.paddingOf(context).bottom
+              : MediaQuery.paddingOf(context).bottom,
+          // top: position.dy,
+          child: Column(
```
Comment (ice-endymion) [inline lib/app/features/chat/messages/views/components/message_reaction_dialog/message_reaction_dialog.dart:90]: Commented code

File: lib/app/features/chat/messages/views/pages/messages_page.dart [hunk 4]
```diff
@@ -45,7 +45,8 @@ class MessagesPage extends ConsumerWidget {
...
...g? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing?How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing? How are you doing?',
```
Comment (ice-endymion) [inline lib/app/features/chat/messages/views/pages/messages_page.dart:49]: Nerd comment 🤓
Maybe it is better to use [lorem ipsum](https://loremipsum.io/generator?n=5&t=p) text and marked that this is mock some how.

File: lib/app/features/chat/messages/views/components/message_reaction_dialog/message_reaction_dialog.dart (hunk 3 above)
Comment (ice-kreios) [reply lib/app/features/chat/messages/views/components/message_reaction_dialog/message_reaction_dialog.dart]: fixed

File: lib/app/features/chat/messages/views/pages/messages_page.dart (hunk 4 above)
Comment (ice-kreios) [reply lib/app/features/chat/messages/views/pages/messages_page.dart]: It looks more good 🌟  thanks for suggestion @ice-alcides !

**Review by ice-endymion — APPROVED** [review]

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**